### DISCUSSION TITLE
Making An API request returns strange error

### DISCUSSION THREAD

**Discussion by atillaersoy** [ludeknovy/jtl-reporter#232] [Q&A] [answered]
Hello Everyone,

While i'm trying to create new test via api, it returns error like below. almost all parameters i pass returns this such error.

Any ideas?

Thanks in advance
![image](https://github.com/ludeknovy/jtl-reporter/assets/9157585/1c7f8fc6-9b68-4d31-b1ba-c747d8e75bea)

**Comment by ludeknovy** [comment]
Hi @atillaersoy what version of BE service do you use?

**Reply by ludeknovy** [reply]
btw it looks like you send the `environment` as form data, but it should be a json.

**Reply by atillaersoy** [reply]
Version is 4.8.4 and i have just added Content-Type:application/json to header, after sending the request it returns me the blank with 500 internal server error. for x-access-token i'm using the one i have created from the admin panel.

![image](https://github.com/ludeknovy/jtl-reporter/assets/9157585/34bae186-8710-4d59-9192-a79a1dfefeb2)

**Reply by ludeknovy** [reply, marked as answer]
do the project and scenario exist? I was getting 500 in case the scenario didn't exist (I need to fix that...)

**Reply by ludeknovy** [reply]
also, if you run the application in docker, you can check the application logs using the following command: `docker logs <service_name>`

**Reply by atillaersoy** [reply]
Well, Definately that was the thing. I thought project and the sceneario can be generated via api. that's why i added these values without creating them. Thanks for the help :)

**Reply by ludeknovy** [reply]
Good. I will create a fix so it would return 404

**Reply by atillaersoy** [reply]
Instead of returning 404, you have some options to create scenarios under the project can be generated on the fly, may be you can put a global settings to make it generate project and scenario on the fly. because as we need this functionality on pipeline, think everybody will have a use on it :D

btw great work, respect.

**Reply by ludeknovy** [reply]
Hi, it works for scenarios the way you have just described - check project settings

**Reply by atillaersoy** [reply]
Yes, that's what i said, but with bad grammar :) I said may be it could be done for projects too.

**Reply by ludeknovy** [reply]
Yes, it could be done. What would be the use case for projects being dynamic?

**Reply by atillaersoy** [reply]
For example, I have a teardown thread group that runs just before the test ends and uploads the jtl file. But, for each project we should go and create new project. instead of that when this request is made it could be created. Also it'll be handy for pipeline integrations for more generic tests.

 For example in our company, if other teams wants to do a simple get request to one page and they don't want to communicate with performance test teams. They can just type the test name and project name on pipeline than get a public report without any communication with the test team.

**Reply by ludeknovy** [reply]
Ok, so that new settings would also need to include to enable to generate the share token.

**Reply by ludeknovy** [reply]
https://github.com/ludeknovy/jtl-reporter/issues/237
